aioredis/sentinel/pool.py

```diff
@@ -232,12 +231,9 @@ async def discover(self, timeout=None):  # TODO: better name?
         # TODO: discovery must be done with some customizable timeout.
         if timeout is None:
             timeout = self.discover_timeout
-        tasks = []
         pools = []
-        for addr in self._sentinels:  # iterate over unordered set
-            tasks.append(self._connect_sentinel(addr, timeout, pools))
-        done, pending = await asyncio.wait(tasks, return_when=ALL_COMPLETED)
-        assert not pending, ("Expected all tasks to complete", done, pending)
+        tasks = [self._connect_sentinel(addr, timeout, pools) for addr in self._sentinels]
+        done = await asyncio.gather(*tasks, return_exceptions=True)
```

Comment:
I think this is incorrect; `done` was a list of tasks and after the change, it is a list of results.
Moreover, `asyncio.wait()` was not deprecated.

Reply:
I'm for moving to using `gather`, but @asvetlov is right, this will now be an array of results instead of tasks. You'll need to update the `for task in done:` loop to reflect that.

Reply:
Ah right; correct if I'm wrong, but I'm basically removing the for loop since it was only executing the future?

@asvetlov Passing in coroutines was deprecated in Python 3.8, hence the referenced issue and PR. The coroutine itself is not deprecated.

Reply:
The passing-coroutines was deprecated in favor of creating tasks for coroutines explicitly and passing them into `asyncio.wait()`.

I'm -0 for the conversion to `gather()` but the current change has an error.

Reply:
@Andrew-Chen-Wang -

I think we can drop the for-loop completely, unless we'd like to pro-actively log any errors.